### DISCUSSION TITLE
[要望][アイディア] レビュー→承認 の機能

### DISCUSSION THREAD

**Discussion by acevif** [increments/qiita-discussions#434] [要望・アイデア・不具合]
レビュー機能はこのような機能です。

目的:

* ユーザーは、承認したユーザーやその数を見ることで、その記事の信頼性を確認・推測することができる
* いいねは感謝も含めた気軽なコミュニケーション / 承認は情報の精査という意味 という棲み分けが明確になり、いいね もしやすくなる

仕様案:

* 記事には、その記事を承認している人たちを表示する
  * 1行に収まる範囲内で、著者の近くに表示。(信頼性が気になる時はその辺を見ると一目でわかるように)
  * ソート順は: 最新バージョンを承認しているフォロイー → 旧バージョンを承認しているフォロイー → 非フォロイー の順
* ユーザーは、自分が詳しい分野の記事の特定のバージョンに対して内容のレビューを行い、よいと思えば承認することができる
* ユーザーは、1つ以上のバージョンの集合としての記事を一気に承認することは**できない**
* レビュー承認は、手軽に取り消すことはできない
  * UIの動線の工夫 (いいねは1クリックで取り消せるが、レビュー承認はそう簡単ではない)
  * レビュー承認の取り消しは、詳細ページなどに表示され、記録が残る
* フォローしているユーザーが承認をした時には、ホームやタイムラインに流れてくる
* 承認をする際、承認理由のレビューコメントをつけることができる
* ユーザーが承認した記事の一覧はユーザーページなどで公開される

**Comment by acevif** [comment]
昨年、LGTMがいいねに戻りました。私は「いいね」とLGTMは意味が異なるため戸惑っていたので、この変更を歓迎しています。

さて、LGTM機能がついていた頃からあたためていた要望なのですが、レビュー機能があると嬉しいです。
少し検索してみて他にdiscussionが見つからなかったため、投稿させていただきます。

**Comment by acevif** [comment]
要望というより思いつきレベルですが、追加でこのようなアイディアも面白いかもしれません:

* レビュー承認理由コメントのデフォルトメッセージがLGTM!
* 「承認」では固いので、「レビュー済み」「LGTM」「お墨付き」などの名称にする
* レビュー依頼
  * ユーザーは他のユーザーに対してレビュー依頼を投げることができる
  * ユーザーは公開前の記事についても、レビュー依頼を投げることができる
  * レビュー依頼受け入れ対象タグを設定することができる
  * レビュー依頼を受任する/しないを設定することができる
* Qiita公認レビュアー
* 匿名の査読 (double anonymized=レビュイーもレビュアーも誰による記事かわからないまま公開前にレビューし意見交換。承認されると公開される)
* 記事の品質が期待に沿わない場合、request changesすることができる
* 編集リクエストに対してレビューを行い、改善提案があればコメントをつけることができる
* 承認数の制限 月に4本までしかレビュー承認できない など

**Reply by tanaka-asami415** [reply]
@acevif 
たくさんのアイディアをありがとうございます！参考にさせていただきます🙏
